src/formatMarkup.js

```diff
@@ -1,31 +1,70 @@
+// @ts-check
+/** @import { DefaultTreeAdapterMap } from "parse5" */
+
 import { parseFragment } from 'parse5';
 
 import { logger } from '@/helpers.js';
 
+const VOID_ELEMENTS = Object.freeze([
+  'area',
+  'base',
+  'br',
+  'col',
+  'embed',
+  'hr',
+  'img',
+  'input',
+  'link',
+  'meta',
+  'param',
+  'source',
+  'track',
+  'wbr'
+]);
+
+const WHITESPACE_DEPENDENT_TAGS = Object.freeze([
+  'a',
+  'pre'
+]);
+
+/**
+ * @param {DefaultTreeAdapterMap["node"]} node
+ * @returns {node is DefaultTreeAdapterMap["element"]}
+ */
+const isElementNode = (node) => Object.prototype.hasOwnProperty.call(node, 'tagName');
+
+/**
+ * @param {DefaultTreeAdapterMap["node"]} node
+ * @returns {node is DefaultTreeAdapterMap["textNode"]}
+ */
+const isTextNode = (node) => node.nodeName === '#text';
+
 /**
  * Uses Parse5 to create an AST from the markup. Loops over the AST to create a formatted HTML string.
  *
- * @param  {string} markup  Any valid HTML
- * @return {string}         HTML formatted to be more easily diffable
+ * @param  {string} markup Any valid HTML
+ * @param  {object} [options] Settings to tweak the formatting
+ * @param  {'html' | 'xhtml' | 'closingTag'} [options.voidElements='xhtml'] How to handle void elements
```

Comment:
Create a JSDocs defintion above this for the options object, so it can be referenced here.

```suggestion
 * @param  {OPTIONS} [options]  Settings to tweak the formatting
```

[JSDocs definition examples](https://github.com/nwutils/create-desktop-shortcuts/blob/main/api-type-definitions.js)